### PR TITLE
Update 101-virtual-network-manager-create-mesh

### PR DESCRIPTION
**Pull request by mbender-ms**
Update network group membership with use of Azure Policy
Remove old directory that was errantly added on initial PR
Updated Readme

### CODE REVIEW

**Comment by mbender-ms** [comment]
#sign-off

File: quickstart/101-virtual-network-manager-create-mesh/main.tf [hunk 1]
```diff
@@ -58,14 +58,56 @@ resource "azurerm_network_manager_network_group" "network_group" {
...
+  prefix = "network-group-policy"
+}
+
+resource "azurerm_policy_definition" "network_group_policy" {
+  name         = "${random_pet.network_group_policy_name.id}"
+  policy_type  = "Custom"
+  mode         = "Microsoft.Network.Data"
+  display_name = "Policy Definition for Network Group"
+
+  metadata = <<METADATA
```
Comment (lonegunmanb) [inline quickstart/101-virtual-network-manager-create-mesh/main.tf:78]: Could we rewrite this heredoc string with a Terraform native object and [`jsonencode`](https://developer.hashicorp.com/terraform/language/functions/jsonencode) function?

File: quickstart/101-virtual-network-manager-create-mesh/main.tf [hunk 2]
```diff
@@ -58,14 +58,56 @@ resource "azurerm_network_manager_network_group" "network_group" {
...
+    {
+      "category": "Azure Virtual Network Manager"
+    }
+  METADATA
+
+  policy_rule = <<POLICY_RULE
```
Comment (lonegunmanb) [inline quickstart/101-virtual-network-manager-create-mesh/main.tf:84]: Could we rewrite this heredoc string with a Terraform native object and [`jsonencode`](https://developer.hashicorp.com/terraform/language/functions/jsonencode) function?

**Review by grayzu — APPROVED** [review]
Thanks for the PR! LGTM

File: quickstart/101-virtual-network-manager-create-mesh/main.tf [hunk 3]
```diff
@@ -58,14 +58,56 @@ resource "azurerm_network_manager_network_group" "network_group" {
...
+        "allOf": [
+          {
+              "field": "type",
+              "equals": "Microsoft.Network/virtualNetworks"
+          },
+          {
+            "allOf": [
+              {
+              "field": "Name",
+              "contains": "vnet"
```
Comment (lonegunmanb) [inline quickstart/101-virtual-network-manager-create-mesh/main.tf]: This prerequisite seems too loose and might interfere other tests in our test subscription. I would suggest a more precise prerequisite, like `match`, and make vnet's prefix random, like:

```hcl
resource "random_string" "prefix" {
  length = 4
  special = false
  upper = false
}

resource "random_pet" "virtual_network_name" {
  prefix = "vnet${random_string.prefix.result}"
}
```

Then the `allof` could be:

```json
{
  "field": "Name",
  "match": "${random_pet.virtual_network_name.id}.+"
}
```

**Review by lonegunmanb — CHANGES_REQUESTED** [review]
Thanks @mbender-ms for the update, some comments.

File: quickstart/101-virtual-network-manager-create-mesh/main.tf (hunk 2 above)
Comment (mbender-ms) [reply quickstart/101-virtual-network-manager-create-mesh/main.tf:84]: I'll pass this on to the Product Team, and see if an engineer is able to put this together.

File: quickstart/101-virtual-network-manager-create-mesh/main.tf (hunk 1 above)
Comment (mbender-ms) [reply quickstart/101-virtual-network-manager-create-mesh/main.tf:78]: I'll pass this on to the Product Team, and see if an engineer is able to put this together.

File: quickstart/101-virtual-network-manager-create-mesh/main.tf (hunk 3 above)
Comment (mbender-ms) [reply quickstart/101-virtual-network-manager-create-mesh/main.tf]: @lonegunmanb Thanks for the suggestion. I implemented the random string addition and the new match using the random_pet name for VNET in azure policy. Works great!

File: quickstart/101-virtual-network-manager-create-mesh/main.tf (hunk 1 above)
Comment (grayzu) [reply quickstart/101-virtual-network-manager-create-mesh/main.tf:78]: This is not something the PG needs to be aware of / change but I think retaining the Azure Policy format in this case is probably better.

**Pull request merged**